tests/src/JIT/HardwareIntrinsics/X86/Sse2/CompareGreaterThan.cs

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace IntelHardwareIntrinsicTest
+{
+    internal static partial class Program
+    {
+        private const int Pass = 100;
+        private const int Fail = 0;
+
+        static unsafe int Main(string[] args)
+        {
+            int testResult = Pass;
+            int testsCount = 21;
+            string methodUnderTestName = nameof(Sse2.CompareGreaterThan);
+
+            if (Sse2.IsSupported)
+            {
+                using (var doubleTable = TestTableSse2<double>.Create(testsCount))
+                using (var intTable = TestTableSse2<int>.Create(testsCount))
+                using (var shortTable = TestTableSse2<short>.Create(testsCount))
+                using (var sbyteTable = TestTableSse2<sbyte>.Create(testsCount))
+                {
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<double>, Vector128<double>, Vector128<double>) value = doubleTable[i];
+                        var result = Sse2.CompareGreaterThan(value.Item1, value.Item2);
+                        doubleTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<int>, Vector128<int>, Vector128<int>) value = intTable[i];
+                        var result = Sse2.CompareGreaterThan(value.Item1, value.Item2);
+                        intTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<short>, Vector128<short>, Vector128<short>) value = shortTable[i];
+                        var result = Sse2.CompareGreaterThan(value.Item1, value.Item2);
+                        shortTable.SetOutArray(result);
+                    }
+
+                    for (int i = 0; i < testsCount; i++)
+                    {
+                        (Vector128<sbyte>, Vector128<sbyte>, Vector128<sbyte>) value = sbyteTable[i];
+                        var result = Sse2.CompareGreaterThan(value.Item1, value.Item2);
+                        sbyteTable.SetOutArray(result);
+                    }
+
+                    CheckMethod<double> checkDouble = (double x, double y, double z, ref double a) =>
+                    {
+                        a = x > y ? BitConverter.Int64BitsToDouble(-1) : 0;
+                        return double.IsNaN(z) ? double.IsNaN(a) : a == z;
```

Comment:
We should be doing bit comparisons for these (at the very least for `a == z`, but possibly for `IsNaN` as well, to ensure payload is properly preserved).

Reply:
Will fix